src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -0,0 +1,242 @@
+{
+  const int *edges = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
+                                                          T8_CMESH_OCC_CURVE_ATTRIBUTE_KEY,
+                                                          gtreeid);
+  const int *faces = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
+                                                          T8_CMESH_OCC_SURFACE_ATTRIBUTE_KEY,
+                                                          gtreeid);
+  
+  /* Compute coordinates via trilinear interpolation */
+  t8_geom_compute_linear_geometry (active_tree_class,
+                                   active_tree_vertices, ref_coords,
+                                   out_coords);
+  
+  /* Sequence of normal vectors to do linear interpolations */
+  int normal_sequence[3][3] = 
+  {
+    {1, 2, 0},
+    {0, 2, 1},
+    {0, 1, 2}
+  };
+  
+  const int
+  t8_edge_vertex_to_tree_vertex[T8_ECLASS_MAX_EDGES][2] =
+  {
+    {0, 1}, {2, 3}, {4, 5}, {6, 7}, {0, 2}, {4, 6}, {1, 3}, {5, 7}, {0, 4}, {1, 5}, {2, 6}, {3, 7}    /* hex */
+  };
+
+  double interpolated_coords[3], param[2], edge_delta[3]= {0, 0, 0}, cur_edge_delta[3], surface_deltas[6][3];
+  gp_Pnt pnt;
+  
+  /* Check each edge for geometry */
+  for (int i = 0; i < 12; ++i)
+  {
+    if (edges[i] >= 0)
+    {
+      /* Interpolate coordinates between edge vertices */
+      for (int j = 0; j < 3; ++j)
+      {
+        interpolated_coords[j] = (1 - ref_coords[i % 3]) * active_tree_vertices[t8_edge_vertex_to_tree_vertex[i][0] * 3 + j]
+                                + ref_coords[i % 3] * active_tree_vertices[t8_edge_vertex_to_tree_vertex[i][1] * 3 + j];
+      }
+
+      /* Interpolate parameters between edge vertices */
+      const double *parameters = (double *) t8_cmesh_get_attribute(cmesh, t8_get_package_id (),
+                                                                  T8_CMESH_OCC_CURVE_PARAMETERS_ATTRIBUTE_KEY + i,
+                                                                  gtreeid);
+      param[0] = (1 - ref_coords[i % 3]) * parameters[0]
+                  + ref_coords[i % 3] * parameters[1];
+      
+      /* Check if calculated parameters are valid */
+      T8_ASSERT(!t8_global_occ_curve[edges[i]].IsNull());
+      double u1, u2;
+      u1 = t8_global_occ_curve[edges[i]]->FirstParameter();
+      u2 = t8_global_occ_curve[edges[i]]->LastParameter();
+      T8_ASSERT((u1 <= param[0] && param[0] <= u2) || (u2 <= param[0] && param[0] <= u1));
+
+      /* Compute displacement between vertex interpolation and curve evaluation with interpolated parameters */
+      t8_global_occ_curve[edges[i]]->D0(param[0], pnt);
+      cur_edge_delta[0] = pnt.X() - interpolated_coords[0];
+      cur_edge_delta[1] = pnt.Y() - interpolated_coords[1];
+      cur_edge_delta[2] = pnt.Z() - interpolated_coords[2];
+      
+      /* Multiply curve displacement with corresponding ref coords */
+      switch (i % 4)
```

Comment:
Why 4?

Reply:
couldn't find the exact commit, but I fixed it